modules/libcom/RTEMS/posix/rtems_init.c

```diff
@@ -0,0 +1,1159 @@
+{
+    int bound = 0;
+    char iName[16];
+    char *name;
+    char *value;
+
+	(void)hook;
+
+    char ifnamebuf[IF_NAMESIZE];
+    sprintf(ifnamebuf, "%s", getPrimaryNetworkInterface());
+
+    while (*env != NULL) {
+        name = strtok(*env,"=");
+        value = strtok(NULL,"=");
+        printf("all out ---> %s = %s\n", name, value);
+        if (!strncmp(name, "interface", 9) && !strcmp(value,  ifnamebuf))
+			 strncpy(iName, value, 16);
+		if (!strncmp(name, "reason", 6) && !strncmp(value, "BOUND", 5)){
+            printf ("Interface %s bounded\n", iName);
+            bound = 1;
+		}
+		if (bound) {
+            // as there is no ntp-support in rtems-libbsd, we call our own client
+            if(!strncmp(name, "new_ntp_servers", 15))
+                strcpy(rtemsInit_NTP_server_ip,value);
+            if(!strncmp(name, "new_host_name", 13))
+                sethostname (value, strlen (value));
+            if(!strncmp(name, "new_tftp_server_name", 20)){
+                //printf(" new_tftp_server_name : %s\n", value);
+                strncpy(rtems_bsdnet_bootp_server_name,value, sizeof(bootp_server_name_init));
+                printf(" rtems_bsdnet_bootp_server_name : %s\n", rtems_bsdnet_bootp_server_name);
+            }
+            if(!strncmp(name, "new_bootfile_name", 20)){
+                //printf(" new_bootfile_name : %s\n", value);
+                strncpy(rtems_bsdnet_bootp_boot_file_name,value, sizeof(bootp_boot_file_name_init));
+                printf(" rtems_bsdnet_bootp_boot_file_name : %s\n", rtems_bsdnet_bootp_boot_file_name);
+            }
+            if(!strncmp(name, "new_rtems_cmdline", 20)){
+                //printf(" new_rtems_cmdline : %s\n", value);
+                strncpy(rtems_bsdnet_bootp_cmdline,value, sizeof(bootp_cmdline_init));
+                printf(" rtems_bsdnet_bootp_cmdline : %s\n", rtems_bsdnet_bootp_cmdline);
+            }
+            // printf("---> %s = %s\n", name, value);
+        }
+		++env;
+	}
+    if (bound)
+        epicsEventSignal(dhcpDone);
+    }
+
+static rtems_dhcpcd_hook dhcpcd_hook = {
+	.name = "ioc boot",
+	.handler = dhcpcd_hook_handler
+};
+
+static void
+default_network_on_exit(int exit_code, void *arg)
+{
+        rtems_printer printer;
+
+        (void)arg;
+
+        rtems_print_printer_printf(&printer);
+        rtems_stack_checker_report_usage_with_plugin(&printer);
+
+        if (exit_code == 0) {
+                puts("*** END OF TEST ***");
+        }
+}
+
+static void
+default_network_set_self_prio(rtems_task_priority prio)
+{
+        rtems_status_code sc;
+
+        sc = rtems_task_set_priority(RTEMS_SELF, prio, &prio);
+        assert(sc == RTEMS_SUCCESSFUL);
+}
+
+static void
+default_network_dhcpcd(void)
+{
+    static const char default_cfg[] = "clientid FHI test client\n";
+    rtems_status_code sc;
+    int fd;
+    int rv;
+    ssize_t n;
+
+    fd = open("/etc/dhcpcd.conf", O_CREAT | O_WRONLY,
```

Comment:
Should `dhcpd.conf` (and maybe `/etc/hosts`) be separate files captured at compile time?  It seems likely that someone will want site configuration in here.  I think the mechanics of `epicsMakeMemFs.pl` could be reused for this.

Reply:
Herewith I added the DHCP_option for "rtems_cmdline" (DHCP option 129).
"
nodhcp6
ipv4only
option ntp_servers
option  rtems_cmdline
option tftp_server_name
option bootfile_name
"
I'll be happy to take another look at epicsMakeMemFs.pl, though.

Reply:
Just to keep you guys up to date.
I am currently working with qemu with xilinx_zynq_a9_qemu as target.
I have re-enabled the mechanism of epicsMakeMemFs.pl. So when the target boots and no DHCP information is received, there is a local directory into which one can "go" into:
...
***** Setting up file system *****
***** Using compiled in file data *****
-> /iocshTestBadArgIndirect.cmd - ok
-> /iocshTestBadArg.cmd - ok
-> /iocshTestSuccessIndirect.cmd - ok
-> /iocshTestSuccess.cmd - ok
***** Initializing NFS *****
check for time registered , C++ initialization ...
***** Preparing EPICS application *****
***** set directory to / *****
chdir("/")
***** Starting EPICS application *****
...
Parallel to the qemu - stuff I am setting up a vde-switch, dnsmask, ntp-server and nfs-server 
to be able to emulate a "normal" environment completely.

Reply:
I'll  take a look at this, but I propose handling it as a separate pull request.  Unless people think it needs to go in with this one?